core/arch/arm/kernel/boot.c

```diff
@@ -1338,6 +1350,17 @@ void __weak boot_init_primary_late(unsigned long fdt)
 		DMSG("Memory tagging %s",
 		     memtag_is_enabled() ?  "enabled" : "disabled");
 
+	/* Check if platform needs NMFI workaround */
+#if CFG_TEE_CORE_LOG_LEVEL >= TRACE_DEBUG
```

Comment:
Is this `#if` needed?

Reply:
My idea was that it is a waste in a way and in normal images there is no such need. Whoever is porting OP-TEE most likely runs with traces enabled and thus would get the notes.

In a way we could even have separate porting check code that tries to probe some things that needs configuration -- if that is doable / viable.

Reply:
You're concerned that the call to `read_sctlr()` in `cpu_nmfi_enabled()` might not be optimized away by the kernel even if the `IMSG()` doesn't do anything?

Reply:
> My idea was that it is a waste in a way and in normal images there is no such need. Whoever is porting OP-TEE most likely runs with traces enabled and thus would get the notes.

I think `IMSG()` already provides a accurate trace level for these messages. I don't neither think the `#if/#endif` in `TRACE_LEVEL`.

As for the optimization view, as this is optee initialization, i guess it's ok reading few more CPU registers in this sequence.

Reply:
OK. Removing those #if's.